core/src/main/java/org/apache/spark/shuffle/sort/ShuffleExternalSorter.java

```diff
@@ -161,6 +161,10 @@ private void writeSortedFile(boolean isLastFile) {
     final ShuffleInMemorySorter.ShuffleSorterIterator sortedRecords =
       inMemSorter.getSortedIterator();
 
+    // If there are no sorted records, so we don't need to create an empty spill file.
+    if (!sortedRecords.hasNext()) {
+      return;
+    }
```

Comment:
@wangjiaochun . Shall we move line 160 ~ 167 to the beginning of this function (line 147)?

Reply:
I think it's better not to do that.Because change the original code style and it don't makes an appreciable difference in readability.